go/vt/vtctl/reparentutil/util_test.go

```diff
@@ -139,6 +139,115 @@ func TestElectNewPrimary(t *testing.T) {
 			},
 			errContains: nil,
 		},
+		{
+			name: "Two good replicas, but one of them backing up so we pick the other one",
+			tmc: &chooseNewPrimaryTestTMClient{
+				// both zone1-101 and zone1-102 are equivalent from a replicaiton PoV, but zone1-102 is taking a backup
+				replicationStatuses: map[string]*replicationdatapb.Status{
+					"zone1-0000000101": {
+						Position:  "MySQL56/3E11FA47-71CA-11E1-9E33-C80AA9429562:1-5",
+						BackingUp: true,
+					},
+					"zone1-0000000102": {
+						Position:  "MySQL56/3E11FA47-71CA-11E1-9E33-C80AA9429562:1-5",
+						BackingUp: false,
+					},
+				},
+			},
+			tolerableReplLag: 50 * time.Second,
+			shardInfo: topo.NewShardInfo("testkeyspace", "-", &topodatapb.Shard{
+				PrimaryAlias: &topodatapb.TabletAlias{
+					Cell: "zone1",
+					Uid:  100,
+				},
+			}, nil),
+			tabletMap: map[string]*topo.TabletInfo{
+				"primary": {
+					Tablet: &topodatapb.Tablet{
+						Alias: &topodatapb.TabletAlias{
+							Cell: "zone1",
+							Uid:  100,
+						},
+						Type: topodatapb.TabletType_PRIMARY,
+					},
+				},
+				"replica1": {
+					Tablet: &topodatapb.Tablet{
+						Alias: &topodatapb.TabletAlias{
+							Cell: "zone1",
+							Uid:  101,
+						},
+						Type: topodatapb.TabletType_REPLICA,
+					},
+				},
+				"replica2": {
+					Tablet: &topodatapb.Tablet{
+						Alias: &topodatapb.TabletAlias{
+							Cell: "zone1",
+							Uid:  102,
+						},
+						Type: topodatapb.TabletType_REPLICA,
```

Comment:
@GuptaManan100 I assume we already have tests that verify that tablets of type `BACKUP` are always excluded?

Reply:
Yes.